src/SoundNotificationsPlayer/index.js

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useRef } from 'react';
```

Comment:
This component seems a bit muddled to me. We now have:

1. Two main effect hooks, for playing sounds under different conditions, each with some bespoke qualifying logic but duplicating a lot of the same 
2. One extra toast component defined inline

It seems like the refactor could go a step further -- a `useNotificationSound` hook which receives a qualifying function to determine the conditions under which a notification plays, and a separate hook (or some other means, I don't have strong feelings on the **how**) which determines when to show the toast.  

Then you can pull the duplicate code such as
```
 setTimeout(() => {
        canPlayNewEventNotificationSound.current = true;
      }, SHOULD_PLAY_DEBOUNCE_MS);
```

and have more legible code handling these sounds, without expanding this individual component file if/whenever we want to have a new sound-triggering condition.

Does this make sense to you?

Reply:
Yes, I like the idea. However I have a proposal for you. Doing those changes here implies me doing double work: the refactor here and then applying the changes in the other ticket I have for playing sounds for new messages. Are you ok if we keep this one with this pattern for now and just focus on feedback regarding the logic of when to play the sound? And I will add a cool pattern in the next PR, which btw, I'm already working on it @JoshuaVulcan